src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -848,22 +848,50 @@ public static int Sum<T>(this ImmutableArray<T> items, Func<T, int> selector)
             return sum;
         }
 
-        internal static Dictionary<K, ImmutableArray<T>> ToDictionary<K, T>(this ImmutableArray<T> items, Func<T, K> keySelector)
+        private static class AccumulatorCache<K>
```

Comment:
right now the ToDictionary helper is called from two layers (IDE and compiler).  In the compiler layer, it's only used in SourceMemberContainerSymbol, and in the IDE it's only used in EnC.  An alternative approach to this would be to just move this helper into a specialized one inside SourceMemberContainerSymbol, and take the other one and put it in EnC.  Then we wouldn't have to have this sort of specialization for the different equality comparer semantics they both have.